pkg/virt-api/webhooks/validating-webhook/admitters/vmclone-admitter.go

```diff
@@ -77,37 +73,7 @@ func (admitter *VirtualMachineCloneAdmitter) Admit(ctx context.Context, ar *admi
 		return webhookutils.ToAdmissionResponseError(err)
 	}
 
-	var causes []metav1.StatusCause
-
-	if newCauses := validateFilters(vmClone.Spec.AnnotationFilters, "spec.annotations"); newCauses != nil {
-		causes = append(causes, newCauses...)
-	}
-	if newCauses := validateFilters(vmClone.Spec.LabelFilters, "spec.labels"); newCauses != nil {
-		causes = append(causes, newCauses...)
-	}
-	if newCauses := validateFilters(vmClone.Spec.Template.AnnotationFilters, "spec.template.annotations"); newCauses != nil {
-		causes = append(causes, newCauses...)
-	}
-	if newCauses := validateFilters(vmClone.Spec.Template.LabelFilters, "spec.template.labels"); newCauses != nil {
-		causes = append(causes, newCauses...)
-	}
-
-	if newCauses := validateSourceAndTargetKind(vmClone); newCauses != nil {
-		causes = append(causes, newCauses...)
-	}
-
-	if newCauses := validateSource(ctx, admitter.Client, vmClone); newCauses != nil {
-		causes = append(causes, newCauses...)
-	}
-
-	if newCauses := validateTarget(vmClone); newCauses != nil {
-		causes = append(causes, newCauses...)
-	}
-
-	if newCauses := validateNewMacAddresses(vmClone); newCauses != nil {
-		causes = append(causes, newCauses...)
-	}
-
+	causes := clonecontroller.GetCauses(vmClone, ctx, admitter.Client)
```

Comment:
Hey @ayushpatil2122!

The whole idea is to execute these functions as part of virt-controller instead of virt-api, but in this implementation the functions are moved yet the execution remains in the same place. Does this solve the "avoid races and enhance the eventual consistency paradigm" problem that stated in the PR description?